ClockDisplay.java

```diff
@@ -36,9 +38,20 @@ public ClockDisplay()
      */
     public ClockDisplay(int hour, int minute)
     {
-        hours = new NumberDisplay(24);
+        hours = new NumberDisplay(12);
         minutes = new NumberDisplay(60);
-        setTime(hour, minute);
+        if(hour >= 12 && hour != 24)
+        {
+            this.isAM = false;
+        } 
+        else
+        {
+            this.isAM = true;
+        }
+        hours.setValue(hour % 12);
```

Comment:
This works but is unnecessary - by definition hours never greater than 12 in this branch